client/modules/sheaf-core/src/index.ts

```diff
@@ -0,0 +1,204 @@
+            if (lnStart === lnEnd) activeLines.add(lnStart - 1)
+            else {
+              const diff = lnEnd - lnStart
+              for (let i = 0; i <= diff; i++) {
+                activeLines.add(lnStart + i - 1)
+              }
+            }
+          }
+          this.activeLines.set(activeLines)
+        }
+      }
+    }))
+
+    this.view = new EditorView({
+      parent,
+      state: EditorState.create({
+        doc,
+        extensions: [
+          ...getExtensions(),
+          ...extensions,
+          this.spellcheckCompartment.of(
+            EditorView.contentAttributes.of({ spellcheck: "false" })
+          ),
+          this.guttersCompartment.of(EditorView.editorAttributes.of({ class: "" })),
+          updateHandler
+        ]
+      })
+    })
+
+    this.refresh()
+  }
+
+  /** The `Text` object of the editor's current state. */
+  get doc() {
+    return this.view.state.doc
+  }
+
+  get scrollTop() {
+    return this.view.scrollDOM.scrollTop
+  }
+
+  set scrollTop(val: number) {
+    this.view.scrollDOM.scrollTop = val
+  }
+
+  /** Destroys the editor.
+   *  Usage of the editor object after destruction is obviously not recommended. */
+  destroy() {
+    this.view.destroy()
+  }
+
+  refresh() {
+    this.state = this.view.state
+    const value = this.doc.toString()
+    this.store.set({
+      doc: this.doc,
+      value: value
+    })
+  }
+
+  /** Returns the scroll-offset from the top of the editor for the specified line. */
+  heightAtLine(line: number) {
+    return this.view.visualLineAt(this.doc.line(line).from).top
+  }
+
+  printTree() {
+    return printTree(syntaxTree(this.state), this.doc.toString())
+  }
+
+  /** Whether or not the browser's spellchecker is enabled for the editor. */
+  set spellcheck(state: boolean) {
+    this.view.dispatch({
+      effects: this.spellcheckCompartment.reconfigure(
+        EditorView.contentAttributes.of({ spellcheck: String(state) })
+      )
+    })
+  }
+
+  /** Whether or not the line-numbers and associated gutter panel is shown. */
+  set gutters(state: boolean) {
+    this.view.dispatch({
+      effects: this.guttersCompartment.reconfigure(
+        EditorView.editorAttributes.of({ class: state ? "" : "hide-gutters" })
+      )
+    })
+  }
+}
+
+export function getExtensions() {
+  return [
+    lineNumbers(),
+    highlightSpecialChars(),
+    history(),
+    foldGutter(),
+    drawSelection(),
+    EditorState.allowMultipleSelections.of(true),
+    indentOnInput(),
+    bracketMatching(),
+    closeBrackets(),
+    highlightSelectionMatches(),
+    autocompletion(),
+    rectangularSelection(),
+    highlightActiveLine(),
+    EditorView.lineWrapping,
+    indentHack,
+    keymap.of([
+      ...closeBracketsKeymap,
+      ...defaultKeymap,
+      ...searchKeymap,
+      ...historyKeymap,
+      ...foldKeymap,
+      ...commentKeymap,
+      ...completionKeymap,
+      { key: "Mod-l", run: openLintPanel, preventDefault: true },
```

Comment:
I thought we decided on something else? Given that Ctrl+L normally selects the address bar for easy URL changing.

Reply:
There really isn't many other safe bindings, and we didn't decide on anything else. IIRC Ctrl + L is a duplicate of another binding, I think Ctrl + K. This is something we'll have to just test and hope is fine, I think.